tests/integration/test_integration.py

```diff
@@ -93,7 +75,16 @@ async def deploy_charm(ops_test, charm: Path) -> None:
             series="focal",
         )
 
-    async def post_deployment_tasks(self, namespace: str) -> str:
+    @pytest.mark.abort_on_fail
+    @pytest.fixture(scope="module")
+    async def build_and_deploy(self, ops_test):
+        ops_test.destructive_mode = False
+        charm = await ops_test.build_charm(".")
+        await self.deploy_charm(ops_test, charm)
+
+    @pytest.mark.abort_on_fail
+    @pytest.fixture(scope="module")
+    async def post_deployment_tasks(self, ops_test) -> str:
```

Comment:
Add typing to `ops_test` and update docstring to reflect argument change.

Reply:
Also should this be a fixture? To me, it looks like a private method.

Reply:
Docstring done. The reason I make it a fixture is for being run just once, I don't want to generate multiple tokens, is there a better way?

Reply:
Now you run it at least twice. First time it's run as a fixture and the second is when you call it directly in the `test_given_no_config_when_post_deployment_tasks_and_authorise_charm_then_status_is_active` (btw, I think it should be authorize, with `z`, not `s`) test, to get the token.
You need post deployment tasks to get the token you'll use to authorize the charm. When you run post installs as a fixture, you're not even fetching this token.

Reply:
- Am I running it twice? Correct me if I'm wrong, I think the point of `@pytest.fixture(scope="module")` is for that function to be run only once per module. Am I calling it here? `vault_token = post_deployment_tasks` is a reference to the fixture, so it's not being run there, and because the return value should be stored from the first and only run.
- Why am I not fetching the token, and how can that be possible if the tests succeed?
- I don't get this "(btw, I think it should be authorize, with z, not s)", the action in the `actions.yaml` is `authorise-charm`, I guess you don't mean that.

Reply:
Damn, my bad, I don't why I saw `vault_token = post_deployment_tasks()`. Sorry.
Either way though, I'd rather make `post_deployment_tasks` a regular method. It's gonna be run only once anyway, because only one test uses it.
About the typo - yes, that's exactly what I meant. It should be `authorize` instead of `authorise`.

Reply:
About the typo, that's not the difference between American and British English? The one you're suggesting is the American, doesn't it?

Reply:
Pytest runs tests in the same order you put them in your test class. Always. This means, you don't need `post_deployment_tasks` to be a fixture. 

You're right about American vs British English :D

Reply:
And you're right about the fixture. Fixed :)